### DISCUSSION TITLE
[Steam Deck] Protontricks fails to install from Discover Store and Konsole. Looking for any insight

### DISCUSSION THREAD

**Discussion by psuedoh** [Matoking/protontricks#298] [Q&A] [answered]
Using a Steam Deck, I've just re-imaged it and then factory reset it for peace of mind. 

When I try installing protontricks from Konsole using this command: `flatpak install flathub com.github.Matoking.protontricks` I get this error msg: `Failed to install com.github.Matoking.protontricks: While pulling app/com.github.Matoking.protontricks/x86_64/stable from remote flathub: While fetching https://dl.flathub.org/repo/objects/99/[long string of chars].filez: [56] Failure when receiving data from the peer`

When I try installing from the Discover store I get this error msg: `Aborted due to failure (While pulling runtime/org.freedesktop.Platform.Compat.i386/x86_64/23.08 from remote flathub: While fetching https://dl.flathub.org/repo/objects/65/[long string of chars].filez: [55] Failed sending data to the peer)`

I know nothing about Linux, so I'll include anything I think could be helpful below ---

- The command flatpak remotes outputs: 

` NAME OPTIONS`

`flathub system`
`flathub-beta system, no-enumerate `

I've tried 
- `flatpak repair` a few times, with "--system" or "--user", or no arguments.
- `flatpak remote-add --if-not-exists flathub https://dl.flathub.org/repo/flathub.flatpakrepo`
- Ensuring that I'm on the Stable system channel
- Restarting after running most commands. 

And I just tried installing Lutris from the Discover store but got a similar error message.

Thanks for reading! Happy to try anything. I've also asked Steam Support but they can't offer much support understandably.

**Comment by psuedoh** [comment]
@Matoking 🙏 it seems like this isn't an issue with protontricks specifically, so apologies you got lumped into this 😆 if you know of a more relevant forum to ask this on I'd appreciate the insight

**Reply by Matoking** [reply, marked as answer]
No problem. :)

The "[56] Failure when receiving data from the peer" error originates from cURL, a library for performing network requests that **a lot** of projects use. There isn't a single reason for why it might occur, so searching for that error message might help. You could check if you have a VPN or proxy active, or if your system clock is incorrect. Or it might be a temporary problem.

You could also try opening the link directly using the web browser to see if that works, or at least presents a different, more precise error.

Also, you could look into Flatpak's issue tracker here:

https://github.com/flatpak/flatpak/issues?q=error+receiving+from+peer

**Comment by psuedoh** [comment]
Thank you Matoking! I confirmed I'm on not a proxy or VPN, and was able to download a .filez file when opening the link in a browser. 

I did stumble on a workaround though! So just in case someone else is trying to play a game on their steam deck ( S.T.A.L.K.E.R Anomaly in my case) and running into the same issue I'll post my steps here:

In Konsole run:

- flatpak install protontricks
- flatpak run app.com.guthub.Matoking.protontricks/x86_64/stable
- protontricks --gui 
(if you're using an SD card you may get a response saying you have to grant permissions, thankfully it gave me exactly what to copy paste. Looks like `flatpak override --user --filesystem=/`. Then I was able to open the GUI!)
